ipn/local.go

```diff
@@ -223,20 +225,46 @@ func (b *LocalBackend) Start(opts Options) error {
 			b.send(Notify{Prefs: prefs})
 		}
 		if newSt.NetMap != nil {
+			changed := true
+
 			b.mu.Lock()
-			if b.netMapCache != nil && b.cmpDiff != nil {
-				s1 := strings.Split(b.netMapCache.Concise(), "\n")
-				s2 := strings.Split(newSt.NetMap.Concise(), "\n")
-				diff := b.cmpDiff(s1, s2)
-				if strings.TrimSpace(diff) != "" {
-					b.logf("netmap diff:\n%v\n", diff)
+			if b.netMapCache != nil {
+				var hash1, hash2 [sha256.Size]byte
+				h1 := sha256.New()
+				err1 := json.NewEncoder(h1).Encode(newSt.NetMap)
+				h1.Sum(hash1[:0])
```

Comment:
Just use the high-level helper https://golang.org/pkg/crypto/sha256/#Sum256 ?

Reply:
I was trying to avoid allocating the full []byte. Maybe I'm being silly, but a lot of the memory-messy code below is skipped on iOS because cmpDiff == nil.

Reply:
I factored this out into a `jsonHash` function.